Add unstar, delete and clone action

xuchunyang opened the pull request:
To delete a user repo, 'delete_repo' permission is required.

xuchunyang committed:
Add unstar and delete action

To delete a user repo, 'delete_repo' permission is required.

xuchunyang committed:
Check unstar and delete API return

xuchunyang committed:
Add clone action

Fix issue #7

Sliim commented:
Thanks! again :)
I did a quick review, looks good :), I think there is two minor improvements:
- Is it possible to have variables definitions before functions definitions?
- The delete action, hmm ok why not, personnaly I will not use this feature :). But I think this action should be the last (sorted actions will be: `browse`, `show`, `clone`, `delete`).

xuchunyang committed:
Adjust order of actions

Put unstar/delete action to the last.

xuchunyang committed:
Place variables definitions before functions definitions

To achieve this, when evaluating these variables definitions forms,
unknown functions can not be evaluated, so I replaced calling
`helm-github-stars-source-init' function (which takes a argument and
return a lambda form) with a lambda form.

xuchunyang commented:
Done.

Sliim commented:
Awesome! Merged!
Thank you :)
